internal/metastore/kv/datacoord/kv_catalog.go

```diff
@@ -136,75 +136,57 @@ func (kc *Catalog) listSegments(ctx context.Context) ([]*datapb.SegmentInfo, err
 		return nil
 	}
 
-	err := kc.MetaKv.WalkWithPrefix(ctx, SegmentPrefix+"/", kc.paginationSize, applyFn)
+	err := kc.MetaKv.WalkWithPrefix(ctx, buildCollectionPrefix(collectionID), kc.paginationSize, applyFn)
 	if err != nil {
 		return nil, err
 	}
 
 	return segments, nil
 }
 
-func (kc *Catalog) parseBinlogKey(key string, prefixIdx int) (int64, int64, int64, error) {
-	remainedKey := key[prefixIdx:]
-	keyWordGroup := strings.Split(remainedKey, "/")
+func (kc *Catalog) parseBinlogKey(key string) (int64, error) {
+	// by-dev/meta/datacoord-meta/binlog/454086059555817418/454086059555817543/454329387504816753/1
+	// ---------------------------------|collectionID      |partitionID       |segmentID         |fieldID
+	keyWordGroup := strings.Split(key, "/")
 	if len(keyWordGroup) < 3 {
```

Comment:
if len(keyWordGroup) !=  8?

is there a chance parseBinlogKey diddn't follow this format?

Reply:
> if len(keyWordGroup) != 8?

The purpose here is solely to retrieve the segmentID, so it suffices to take the second-to-last one.

> is there a chance parseBinlogKey diddn't follow this format?

There are no exceptions; currently, the format for all binlog path keys is `${rootPath}/meta/datacoord-meta/${log-type}/${collectionID}/${partitionID}/${segmentID}/${fieldID}`, including binlog, deltalog, statslog, and bm25log.